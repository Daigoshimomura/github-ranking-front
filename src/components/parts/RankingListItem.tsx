import React from "react";
import { UserMinimumInformation } from "../../external/data/UserMinimumInformation";
import {
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText
} from "@material-ui/core";

interface RankingListItemProps {
  rank: Number;
  user: UserMinimumInformation;
  itemClickCallback: Function;
}

export class RankingListItem extends React.Component<RankingListItemProps, {}> {
  constructor(props: RankingListItemProps) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.itemClickCallback(this.props.user.userId);
  }

  render() {
    return (
      <ListItem onClick={this.handleClick} style={{ cursor: "pointer" }}>
        <ListItemAvatar>
          <Avatar
            alt={this.props.user.userId}
            src={this.props.user.avatarUrl}
          />
        </ListItemAvatar>
        <ListItemText primary={this.props.rank} />
        <ListItemText primary={this.props.user.userId} />
      </ListItem>
    );
  }
}
