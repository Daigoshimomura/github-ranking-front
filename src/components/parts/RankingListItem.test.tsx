import { mount } from "enzyme";
import React from "react";
import { RankingListItem } from "./RankingListItem";
import { ListItem } from "@material-ui/core";
import renderer from "react-test-renderer";

const user = {
  userId: "userId",
  score: 0,
  mainLanguage: "mainLanguage",
  avatarUrl: "avatarUrl",
  githubUrl: "githubUrl"
};
const testMock = jest.fn();

describe("RankingListItem", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <RankingListItem rank={1} user={user} itemClickCallback={testMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("レンダリングの確認", () => {
    const wrapper = mount(
      <RankingListItem rank={1} user={user} itemClickCallback={testMock} />
    );
    wrapper.find(ListItem);
    expect(wrapper.find(ListItem).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認", () => {
    const wrapper = mount(
      <RankingListItem rank={1} user={user} itemClickCallback={testMock} />
    );
    expect(wrapper.prop("rank")).toBe(1);
    expect(wrapper.prop("user").userId).toBe("userId");
    expect(wrapper.prop("itemClickCallback")).toBe(testMock);
  });
  it("クリック時にコールバックが実行されることを確認", () => {
    const wrapper = mount(
      <RankingListItem rank={1} user={user} itemClickCallback={testMock} />
    );
    wrapper.find(ListItem).simulate("click");
    expect(testMock).toHaveBeenCalled();
  });
});
