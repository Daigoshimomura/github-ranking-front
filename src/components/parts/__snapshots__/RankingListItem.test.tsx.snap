// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingListItem snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="MuiListItemAvatar-root"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle"
    >
      <img
        alt="userId"
        className="MuiAvatar-img"
        src="avatarUrl"
      />
    </div>
  </div>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      1
    </span>
  </div>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      userId
    </span>
  </div>
</li>
`;
